lib/App/Netdisco/Worker/Plugin/Arpnip/Nodes.pm

```diff
@@ -63,4 +66,78 @@ sub get_arps {
   return $resolved_ips;
 }
 
+register_worker({ phase => 'main', driver => 'cli' }, sub {
+    my ($job, $workerconf) = @_;
+
+    my $device = $job->device;
+    my ($ssh, $selected_auth) = App::Netdisco::Transport::CLI->session_for($device->ip, "sshcollector");
+
+    if (get_arps_cli($device, $ssh, $selected_auth)){
+      my $now = 'to_timestamp('. (join '.', gettimeofday) .')';
+      $device->update({last_arpnip => \$now});
+      my $endmsg = "Ended arpnip cli for $device"; 
+
+      if ($selected_auth->{'snmp_arpnip_also'}){
+        $endmsg .= ", now running arpnip due to snmp_arpnip_also"; 
+        info sprintf " [%s] arpnip cli - $endmsg", $device->ip;
+        return Status->info($endmsg);
+      }else{
+        info sprintf " [%s] arpnip cli - $endmsg", $device->ip;
+        return Status->done($endmsg);
+      }
+    }else{
+      Status->defer("arpnip cli failed");
+    }
+
+});
+
+sub get_arps_cli {
+  my ($device, $ssh, $selected_auth) = @_;
+
+
+  unless ($ssh){
+    my $msg = "could not connect to $device with SSH, deferring job"; 
+    warning sprintf " [%s] arpnip cli - %s", $device->ip, $msg;
+    return undef;
+  }
+
+  my $class = "App::Netdisco::SSHCollector::Platform::".$selected_auth->{platform};
+  debug sprintf " [%s] arpnip cli - delegating to platform module %s", $device->ip, $class;
+
+  my $load_failed = 0;
+  try {
+    Module::Load::load $class;
+  } catch {
+    warning sprintf " [%s] arpnip cli - failed to load %s: %s", $device->ip, $class, substr($_, 0, 50)."...";
+    $load_failed = 1;
+  };
+  return undef if $load_failed;
+
+  my $platform_class = $class->new();
+  my $arpentries = [ $platform_class->arpnip($device->ip, $ssh, $selected_auth) ];
+
+  if (not scalar @$arpentries) {
+    warning sprintf " [%s] WARNING: no entries received from device", $device->ip;
+  }
+
+  hostnames_resolve_async($arpentries);
+
+  foreach my $arpentry ( @$arpentries ) {
+
+    # skip broadcast/vrrp/hsrp and other weirdos
+    next unless check_mac( $arpentry->{mac} );
+
+    debug sprintf ' [%s] arpnip cli - stored entry: %s / %s / %s',
+    $device->ip, $arpentry->{mac}, $arpentry->{ip}, 
+    $arpentry->{dns} if defined $arpentry->{dns};
+    store_arp({
+        node => $arpentry->{mac},
+        ip => $arpentry->{ip},
+        dns => $arpentry->{dns},
+      });
+  }
+
+  return 1;
```

Comment:
is that return needed here? or do we want to return resolved addresses like the snmp version does?